src/context/SyntheticsStateContext/selectors/tradeboxSelectors/selectTradeboxAvailableMarketOptions.ts

```diff
@@ -148,5 +180,82 @@ export const selectTradeboxAvailableMarketOptions = createSelector((q) => {
     }
   }
 
+  for (const liquidMarket of liquidMarkets) {
+    const marketIncreasePositionAmounts = getMarketIncreasePositionAmounts(q, liquidMarket.marketTokenAddress);
+    if (!marketIncreasePositionAmounts) {
+      continue;
+    }
+
+    const positionFeeBeforeDiscount = getFeeItem(
+      marketIncreasePositionAmounts.positionFeeUsd.add(marketIncreasePositionAmounts.feeDiscountUsd).mul(-1),
+      marketIncreasePositionAmounts.sizeDeltaUsd
+    );
+
+    const priceImpactDeltaUsd = getCappedPositionImpactUsd(
+      liquidMarket,
+      marketIncreasePositionAmounts.sizeDeltaUsd,
+      isLong
+    );
+
+    const { acceptablePriceDeltaBps } = getAcceptablePriceByPriceImpact({
+      isIncrease: true,
+      isLong,
+      indexPrice: getMarkPrice({ prices: indexToken.prices, isLong, isIncrease: true }),
+      priceImpactDeltaUsd: priceImpactDeltaUsd,
+      sizeDeltaUsd: marketIncreasePositionAmounts.sizeDeltaUsd,
+    });
+
+    const openFees = positionFeeBeforeDiscount!.bps.add(acceptablePriceDeltaBps);
+
+    if (!result.minOpenFeesBps || openFees.gt(result.minOpenFeesBps)) {
+      result.minOpenFeesBps = openFees;
+      result.minOpenFeesAvailableMarketAddress = liquidMarket.marketTokenAddress;
+    }
+  }
+
   return result;
 });
+
+function getMarketIncreasePositionAmounts(q: QueryFunction<SyntheticsState>, marketAddress: string) {
```

Comment:
I could not come up with a better solution, other solution i had it to make the whole getTradeFee pipeline parametrized, this would require a lot of code (